dist/tools/zsh-completion/_riot

```diff
@@ -0,0 +1,89 @@
+function _boards {
+    local -a _boards_available
+    local -a _board_dirs
+    _board_dirs=( ${(s[ ])EXTERNAL_BOARD_DIRS} )
+    local _repo_root
+    if git rev-parse --is-inside-work-tree &> /dev/null; then
+        local _repo_root="$(git rev-parse --show-toplevel)"
+        _board_dirs+=("$_repo_root/boards")
+    fi
+    for dir ("$_board_dirs[@]") \
+        _boards_available+=($(ls $dir | grep -v common))
+
+    _describe 'board' _boards_available
+}
+
+function _toolchains {
+    local -a _toolchains_available=(
+        "gnu"
+        "llvm"
+    )
+
+    _describe 'toolchain' _toolchains_available
+}
+
+function _bools {
+    local -a _bool_vals=(
+        "0"
+        "1"
+    )
+
+    _describe 'bool' _bool_vals
+}
+
+function _riot {
+    local -a _std_targets=(
```

Comment:
There is an `make info-toolchains-supported` that would be ideal for this. However, this will only work within an application folder. E.g. when typing `make TOOLCHAIN=llvm -C examples/default` in the root of the repository, `make info-toolchains-supported` won't work. (Even if we would parse the cmdline and search for `-C <dir_to_app>`, this might be added as last component and is not available when completing.)

This hardcoded list is less fragile in this regard.

Reply:
`make info-toolchains-supported` just gives the toolchains right? I meant a unique source of truth for supported makefile commands, sth similar to what is given with `make help`. But I do see your point about the missing application folder.